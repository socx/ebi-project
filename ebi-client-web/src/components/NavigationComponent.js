import React from 'react';
import Container from 'react-bootstrap/Container';
import Nav from 'react-bootstrap/Nav';
import Navbar from 'react-bootstrap/Navbar';
import NavDropdown from 'react-bootstrap/NavDropdown';


const options = [
  { label: '25%', value: .25 },
  { label: '50%', value: .5 },
  { label: '75%', value: .75 },
  { label: '100%', value: 1 },
];

const NavigationComponent = ({selectedZoomChanged, setIsDemoModalVisible}) => {
  return (
    <>
      <Navbar variant="dark" bg="dark" expand="lg">
        <Container fluid>
          <Navbar.Brand href="#home">
            <img
              data-testid="navbar-brand-image"
              alt=""
              src={require('./../images/logo.svg')}
              width="30"
              height="30"
              className="d-inline-block align-top"
            />{' '}
            Family Tree App
          </Navbar.Brand>
          <Navbar.Toggle aria-controls="navbar-dark" />
          <Navbar.Collapse id="navbar-dark">
            <Nav>
              <Nav.Link
                onClick={()=> setIsDemoModalVisible(true) }
              >
                Help
              </Nav.Link>
              <NavDropdown
                id="nav-dropdown-dark"
                title="Zoom"
                menuVariant="dark"
                onSelect={selectedZoomChanged}
              >
                {options.map(({ value, label}) => {
                  return <NavDropdown.Item key={value} eventKey={value}>{label}</NavDropdown.Item>
                })}
                <NavDropdown.Divider />
                <NavDropdown.Item key="0.1" eventKey="0.1">Fit to Screen</NavDropdown.Item>
              </NavDropdown>
            </Nav>
          </Navbar.Collapse>
        </Container>
      </Navbar>
    </>
  )
}

export default NavigationComponent
