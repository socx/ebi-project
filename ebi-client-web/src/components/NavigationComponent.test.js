import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import NavigationComponent from './NavigationComponent';

describe('NavigationComponent', () => { 
  test('renders the component', () => { 
    const container = render(<NavigationComponent />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('should display Brand', () => { 
    const { getByTestId, getByText } = render(<NavigationComponent/>);
    expect(getByText('Family Tree App')).toBeInTheDocument();
    expect(getByTestId('navbar-brand-image')).toBeInTheDocument();
  });

  test('should display nav items', () => {
    const { getByText } = render(<NavigationComponent/>);
    expect(getByText('Help')).toBeInTheDocument();
    expect(getByText('Zoom')).toBeInTheDocument();
  });

  test('should display sub-nav items when Zoom is clicked', () => {
    const { getByText } = render(<NavigationComponent />);
    fireEvent.click(getByText('Zoom'));
    expect(getByText('25%')).toBeInTheDocument();
    expect(getByText('50%')).toBeInTheDocument();
    expect(getByText('75%')).toBeInTheDocument();
    expect(getByText('100%')).toBeInTheDocument();
    expect(getByText('Fit to Screen')).toBeInTheDocument();
  });
   
});