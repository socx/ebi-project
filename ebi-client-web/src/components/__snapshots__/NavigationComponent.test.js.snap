// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationComponent renders the component 1`] = `undefined`;
